sam/firestore.tf

```diff
@@ -0,0 +1,46 @@
+# Google project for Sam Firestore
+resource "google_project" "sam-firestore" {
+  count = var.enable && contains(["default", "preview_shared"], var.env_type) ? 1 : 0
```

Comment:
Would it be easier to make a local for this expression since it is repeated alot?

Reply:
What's in the `contains` is different for different resources base don which types of envs they should be created in. I guess I can make locals for all of the different `var.enable && contains([***], var.env_type)` combinations? Not sure how much cleaner that is.

Reply:
Ah I thought they would all evaluate to the same thing within a particular invocation of the module. If that isn't the case then yeah I guess there isn't much benefit to doing this.